src/assets/src/hooks/useOneTrust.ts

```diff
@@ -0,0 +1,50 @@
+import { useState, useEffect } from 'react';
+
+declare global {
+    interface Window {
+      OnetrustActiveGroups?: string;
+      OptanonWrapper?: () => void;
+    }
+  }
+
+export const useOneTrust = (): [boolean, string, (callback: () => void) => void] => {
+    const src = 'https://cdn.cookielaw.org/consent/03e0096b-3569-4b70-8a31-918e55aa20da/otSDKStub.js'
+    const dataDomainScript ='03e0096b-3569-4b70-8a31-918e55aa20da'
+    const [oneTrustActiveGroups, setOneTrustActiveGroups] = useState("");
+    const [loaded, setLoaded] = useState(false)
+
+    useEffect(() => {
+        const script = document.createElement('script');
+        script.src = src;
+        script.type = 'text/javascript';
+        if (dataDomainScript) script.dataset.domainScript = dataDomainScript;
+
+        script.onload = () => {
+            setLoaded(true);
+        }
+        script.onerror = () => {
+            setLoaded(false);
+            console.error(`Failed to load script: ${src}`);
+        }
+
+        document.head.appendChild(script);
+
+        return () => {
+            if (document.head.contains(script)) {
+                document.head.removeChild(script);
+            }
+        };
+    }, []);
+
+    useEffect(() => {
+        if (window.OnetrustActiveGroups) {
+            setOneTrustActiveGroups(window.OnetrustActiveGroups);
+        }
+    }, [window.OnetrustActiveGroups]);
```

Comment:
I don't see anything wrong in having multiple useEffects. As long as it work without error its fine.

https://stackoverflow.com/a/54004148/724238

 Can you also share what these useEffects are doing? I would also leave some comment in code

Reply:
I'll try to leave more clear comments in the code in my next commit. The effects are trying to mimic the Javascript flow listed in the umich cookie instructions (see [my comment](https://github.com/tl-its-umich-edu/remote-office-hours-queue/pull/518#issuecomment-2192068882) ) while keeping the OnetrustActiveGroups state updated, but I hope I'm not causing into too many effect runs that aren't necessary compared to the purely JS script-tag method laid out in the instructions

Reply:
I went ahead and simplified useOneTrust to just export the JS code we needed to run, without useEffects which was complicating it.